### PR TITLE
Log libc version in our "OS info logging test"

### PR DESCRIPTION
**Pull request by danmosemsft**
Add logging of glibc version to our special test that logs the configuration of the machines it runs on.

Note I'm putting the code in PlatformDetection.NonNetfx.cs so it runs on Windows as well -- this will help demonstrate it's safe on all platforms CoreFX tests on, to help the case for adding it to the CLI as telemetry (see https://github.com/dotnet/corefx/issues/24522)

### CODE REVIEW

**Comment by danmosemsft** [comment]
cc @stephentoub

**Comment by danmosemsft** [comment]
NetFX doesn't have PtrToStringUTF8.

File: src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NetFx.cs [hunk 1]
```diff
@@ -24,6 +24,9 @@ public static partial class PlatformDetection
 
         public static bool IsNetfx471OrNewer => GetFrameworkVersion() >= new Version(4, 7, 1);
 
+        public static string LibcRelease { get { return ""; } }
+        public static string LibcVersion { get { return ""; } }
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NetFx.cs]: Nit: `public static string LibcRelease => "";`

File: src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs [hunk 2]
```diff
@@ -12,5 +14,52 @@ public static partial class PlatformDetection
         public static bool IsNetfx462OrNewer => false;
         public static bool IsNetfx470OrNewer => false;
         public static bool IsNetfx471OrNewer => false;
+
+
+        [DllImport("libc", ExactSpelling = true, CharSet = CharSet.Ansi, CallingConvention = CallingConvention.Cdecl)]
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs]: FWIW, Ansi is the default

File: src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs [hunk 3]
```diff
@@ -12,5 +14,52 @@ public static partial class PlatformDetection
...
+        /// If the libc is musl, currently returns empty string.
+        /// Otherwise returns "Glibc not found".
+        /// </summary>
+        public static string LibcRelease
+        {
+            get
+            {
+                try
+                {
+                    return Marshal.PtrToStringUTF8(gnu_get_libc_release());
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs]: Do we not build this lib on netfx?

File: src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs [hunk 4]
```diff
@@ -12,5 +14,52 @@ public static partial class PlatformDetection
...
+                }
+                catch (Exception e) when (e is DllNotFoundException || e is EntryPointNotFoundException)
+                {
+                    return "Glibc not found";
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs]: Nit: Glibc => glibc

File: src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs [hunk 5]
```diff
@@ -12,5 +14,52 @@ public static partial class PlatformDetection
...
+                }
+            }
+        }
+
+        /// <summary>
+        /// If gnulibc is available, returns the version, such as "2.22".
+        /// If the libc is musl, currently returns empty string. (In future could run "ldd -version".)
```
Comment (stephentoub) [inline src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs]: Is this still true?  That suggests musl exposes gnu_get_libc_version and it returns an empty string?  The catch block below returns "Glibc not found".

File: src/System.Runtime.InteropServices.RuntimeInformation/tests/DescriptionNameTests.cs [hunk 6]
```diff
@@ -25,6 +25,19 @@ public void DumpRuntimeInformationToConsole()
...
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Unix interop")]
+        public void DumpRuntimeInformationToConsole2()
+        {
+            // Not really a test, but useful to dump to the log to
+            // sanity check that the test run or CI job
+            // was actually run on the OS that it claims to be on
+            string lcr = PlatformDetection.LibcRelease;
+            string lcv = PlatformDetection.LibcVersion;
+
+            Console.WriteLine($@"LibcRelease={lcr} LibcVersion={lcv}");
```
Comment (stephentoub) [inline src/System.Runtime.InteropServices.RuntimeInformation/tests/DescriptionNameTests.cs]: How about just adding this to the DumpRuntimeInformationToConsole test guarded by an IsFullFramework test?  That way all of the information will be clumped together in the log.

File: src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs (hunk 3 above)
Comment (danmosemsft) [reply src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs]: It's implemented in corelib, so no. We don't have it in an OOB.

Comment (stephentoub) [reply src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.NonNetFx.cs]: Ah, I thought we'd ported it. I guess not yet.

**Comment by danmosemsft** [comment]
OSX10.12 finished clean although check mark isn't shown here

**Pull request merged**